lib/interceptor.js

```diff
@@ -395,9 +395,12 @@ Interceptor.prototype.matchIndependentOfBody = function matchIndependentOfBody(
   }
 
   const checkHeaders = function(header) {
-    return (
-      options.getHeader &&
-      common.matchStringOrRegexp(options.getHeader(header.name), header.value)
+    if (_.isFunction(header.value)) {
```

Comment:
This looks like a bug. When this condition is true, the function returns without executing `common.matchStringOrRegexp()`.